sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -4510,4 +4510,15 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
   private def alterViewTypeMismatchHint: Option[String] = Some("Please use ALTER TABLE instead.")
 
   private def alterTableTypeMismatchHint: Option[String] = Some("Please use ALTER VIEW instead.")
+
+  /**
+   * Create a TimestampAdd expression.
+   */
+  override def visitTimestampadd(ctx: TimestampaddContext): Expression = withOrigin(ctx) {
+    val arguments = Seq(
+      Literal(ctx.unit.getText),
```

Comment:
@MaxGekk I think this indicates the unit parameter must be foldable?

Reply:
This is not the single enter point for `timestampadd`, so, the **must** word is not applicable. See the example above https://github.com/apache/spark/pull/35502#discussion_r810112314, how it should work if `the unit parameter **must** be foldable`. BTW, I will open an JIRA to implement the optimization when `unit` is foldable.

Reply:
Ah I see, then I think the current implementation is fine.